### PR TITLE
Replace static extensions with params

### PR DESCRIPTION
**Pull request by YANG-DB**
### Description
Currently both [FlintJob](https://github.com/opensearch-project/opensearch-spark/blob/main/spark-sql-application/src/main/scala/org/apache/spark/sql/FlintJob.scala) and [SQLJob](https://github.com/opensearch-project/opensearch-spark/blob/main/spark-sql-application/src/main/scala/org/opensearch/sql/SQLJob.scala) are explicitly hard coded using the `FlintSparkExtension` without allowing the job submitter to configure any alternative

### Issues Resolved
Completely remove `extension-list` parameter which will be given to the spark-config as part of the job submission and not be present in the Job input params

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark-sql-application/README.md [hunk 1]
```diff
@@ -52,7 +54,7 @@ aws emr-serverless start-job-run \
     --application-id <application-id> \
     --execution-role-arn <execution-role>  \
     --job-driver '{"sparkSubmit": {"entryPoint": "<flint-job-s3-path>", \
-      "entryPointArguments":["'<sql-query>'", "<result-index>", "<data-source-name>"], \
+      "entryPointArguments":["'<spark extensions>'","'<sql-query>'", "<result-index>", "<data-source-name>"], \
```
Comment (dai-chen) [inline spark-sql-application/README.md:57]: `<data-source-name>` seems moved to `spark.flint.datasource.name`. Can we pass this by Spark conf as well?

Comment (vamsi-amazon) [reply spark-sql-application/README.md:57]: How do we decide if it should be entry point arguments or SparkSubmitParameters. There is one more extension config in SparkSubmitParameters. Which one should we use?

Comment (YANG-DB) [reply spark-sql-application/README.md:57]: IMO the config based features such as:
 - spark extensions
 - data-source-name
 should be part of the in the entry-point.
Other parameters which are more data-driven:
 - query 
 - index names
 are more often changing and should arrive via the submit args - WDYT ?

Comment (penghuo) [reply spark-sql-application/README.md:57]: spark.sql.extensions is a Spark conf. we could remove set in FlintJob and SQLJob and depend on plugin conf it.

Comment (YANG-DB) [reply spark-sql-application/README.md]: I've removed all the `spark extensions` arguments both in FlintJob and SQLJob

**Review by vamsi-amazon — APPROVED** [review]

**Review by penghuo — APPROVED** [review]

**Pull request merged**